prowler/providers/aws/aws_provider.py

```diff
@@ -850,8 +850,7 @@ def assume_role(
                 mfa_info = self.__input_role_mfa_token_and_code__()
                 assume_role_arguments["SerialNumber"] = mfa_info.arn
                 assume_role_arguments["TokenCode"] = mfa_info.totp
-
-            sts_client = create_sts_session(session, AWS_STS_GLOBAL_ENDPOINT_REGION)
```

Comment:
This is the issue here, we were always assuming a role with the Global STS Endpoint Region which is `us-east-1`.